osquery/CMakeLists.txt

```diff
@@ -113,44 +114,66 @@ ADD_OSQUERY_LIBRARY(osquery_amalgamation ${AMALGAMATION})
 add_library(libosquery_additional STATIC ${OSQUERY_ADDITIONAL_SOURCES})
 target_link_libraries(libosquery_additional ${OSQUERY_ADDITIONAL_LINKS})
 set_target_properties(libosquery_additional PROPERTIES OUTPUT_NAME osquery_additional)
-install(TARGETS libosquery_additional DESTINATION lib)
 
-# Optionally build a shared library.
-add_library(libosquery SHARED EXCLUDE_FROM_ALL main/empty.cpp)
-TARGET_OSQUERY_LINK_WHOLE(libosquery libosquery_basic)
-TARGET_OSQUERY_LINK_WHOLE(libosquery libosquery_additional)
-set_target_properties(libosquery PROPERTIES OUTPUT_NAME osquery)
-
-# Include the public API includes.
-install(DIRECTORY "${CMAKE_SOURCE_DIR}/include/" DESTINATION include)
+# Include the public API includes if make devel.
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/include/" DESTINATION include COMPONENT devel OPTIONAL)
 
 add_executable(shell main/shell.cpp)
 TARGET_OSQUERY_LINK_WHOLE(shell libosquery_basic)
 TARGET_OSQUERY_LINK_WHOLE(shell libosquery_additional)
 set_target_properties(shell PROPERTIES OUTPUT_NAME osqueryi)
 set_target_properties(shell PROPERTIES COMPILE_FLAGS ${OS_COMPILE_FLAGS})
-install(TARGETS shell DESTINATION bin)
+install(TARGETS shell RUNTIME DESTINATION bin COMPONENT main)
 
 add_executable(daemon main/daemon.cpp)
 TARGET_OSQUERY_LINK_WHOLE(daemon libosquery_basic)
 TARGET_OSQUERY_LINK_WHOLE(daemon libosquery_additional)
 set_target_properties(daemon PROPERTIES OUTPUT_NAME osqueryd)
 set_target_properties(daemon PROPERTIES COMPILE_FLAGS ${OS_COMPILE_FLAGS})
-install(TARGETS daemon DESTINATION bin)
+install(TARGETS daemon RUNTIME DESTINATION bin COMPONENT main)
 
 add_executable(run main/run.cpp)
 TARGET_OSQUERY_LINK_WHOLE(run libosquery_basic)
 TARGET_OSQUERY_LINK_WHOLE(run libosquery_additional)
 set_target_properties(run PROPERTIES COMPILE_FLAGS ${OS_COMPILE_FLAGS})
-install(TARGETS run DESTINATION bin)
 
-# packages
-if(LINUX)
+# make devel
+add_custom_target(devel
+  COMMAND ${CMAKE_COMMAND}
+    -D COMPONENT=devel
+    -P cmake_install.cmake
+    DEPENDS libosquery_basic
+)
+add_dependencies(devel libosquery_basic)
+
+# make install config files
+install(FILES "${CMAKE_SOURCE_DIR}/tools/deployment/osquery.example.conf"
+  DESTINATION "/var/osquery/" COMPONENT main)
+if(APPLE)
+  install(FILES "${CMAKE_SOURCE_DIR}/tools/deployment/com.facebook.osqueryd.plist"
+    DESTINATION "/var/osquery/" COMPONENT main)
+else()
+  install(PROGRAMS "${CMAKE_SOURCE_DIR}/tools/deployment/osquery.initd"
+    DESTINATION "/etc/init.d/" RENAME "osquery" COMPONENT main)
+endif()
+
+# make package
+if(APPLE)
+  cmake_policy(SET CMP0037 OLD)
+  add_custom_target(
+    package
+    ./tools/deployment/make_osx_package.sh
```

Comment:
do we want to build with -n by default? or do we want to make -n the default if that's the case?